### PR TITLE
(WIP) - Support Time in query

### PR DESCRIPTION
**Pull request by adube**
This PR introduces the support of TIME in the querent service.  On WMS, the `TIME` parameter is used. On WFS requests, it is added to any existing filter on the data source, which gets converted to WFS filter in the end.

Note that the time filter works differently than the filters defined in the Filter tool.  A time filter is set directly on the data source.  It can be changed using the widgets that are drawn in the layer tree.  Those used to set the property directly to the layer.  Now, they set the time value in the data source instead.

A data source that has such a time filter can't be combined to other data source when a query occurs.

## Moment.js

This PR also introduces the Moment.js library, which is used to calculate the range of a DURING filter when we only have 1 time value.  MapServer understands it when we give it as `TIME` property in WMS, but in WFS we need to create it manually.  When we have a 2 time values (start, end), those are used "as-is".

The library has been added to all apps, plus all examples that feature either a map query tool or filter tool.

## Todo

 * [x] Support WMS requests properly, i.e. add `TIME` param in WMS GetFeatureInfo requests
 * [x] Support WFS requests properly, i.e. add them to the filter
 * [ ] Review
 * [ ] Use an official version of OL (currently using @fredj 's branch)

### CODE REVIEW

**Comment by adube** [comment]
Not officially ready for review, but you can begin one if you want.  You can also try it (locally).

**Comment by adube** [comment]
Ready for review!

**Comment by adube** [comment]
Rebased onto master.

**Comment by fredj** [comment]
@adube I can open a PR in OpenLayers if you thing that the during filter is ready

**Comment by adube** [comment]
From what I've seen so far, I'd say that this is ready.  I would feel safer if someone else could also test this.

**Comment by adube** [comment]
I fixed some errors reported by Travis.  How about what's remaining?  Would someone tell me what's wrong?  Thanks.

**Comment by sbrunner** [comment]
I think it's clear (when we found where it is :-) ):
https://travis-ci.org/camptocamp/ngeo/jobs/229993021#L363
https://travis-ci.org/camptocamp/ngeo/jobs/229993025#L1557

**Comment by fredj** [comment]
@adube to search the cause of an error in travis, I search (using the browser search) for `with 2` (or `with 1` or `error 1`)

**Comment by adube** [comment]
Thanks, but that's not what I meant.  I'm sorry, I should have made it more clear: I don't understand the errors.  How can I fix them?

File: .eslintrc.yaml [hunk 1]
```diff
@@ -21,3 +21,4 @@ globals:
     gmfapp: false
     inject: false
     jsts: false
+    moment: false
```
Comment (rbovard) [inline .eslintrc.yaml]: Missing new line (see https://travis-ci.org/camptocamp/ngeo/jobs/229993021#L363)

**Comment by sbrunner** [comment]
Search for `jsts.min.js` in the `Makefile` and you will see what's missing for `moment.min.js` :-)

**Comment by adube** [comment]
Thank you very much.

**Comment by adube** [comment]
Rebased onto master.

**Comment by sbrunner** [comment]
The new version of OpenLayers don't works anymore with node 4 than you should remove this:
https://github.com/camptocamp/ngeo/blob/master/.travis.yml#L26-L30

**Comment by sbrunner** [comment]
And the `desktop` and `desktop_alt` will have an assertion, can you reproduce that on debug mode?

**Comment by adube** [comment]
Thanks for pointing those out.  I made 2 new fixes.

**Comment by fredj** [comment]
thanks @adube

**Comment by adube** [comment]
Thanks @fredj.  I'm surprised to see this merged with the use of you ol3 branch.  Would you please validate that's what you intended to do?

Many thanks,
Alexandre

**Comment by fredj** [comment]
yes, that's intended; so we can test it more easily. I'm creating an issue to not forget to switch to openlayers in package.json once the pull request is merged

**Comment by adube** [comment]
Understood, thanks.

**Pull request merged**